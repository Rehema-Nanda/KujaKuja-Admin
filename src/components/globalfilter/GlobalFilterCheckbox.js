import React from "react";

import { Input, Label, FormGroup } from "reactstrap";
import PropTypes from "prop-types";

export default class GlobalFilterCheckbox extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            checked: props.selectedItems.indexOf(props.item.id) !== -1,
        };

        this.onChange = this.onChange.bind(this);
        this.label = this.label.bind(this);
    }

    onChange = () => {
        const { selectItemHandler, item, selectItemHandlerArgs } = this.props;
        const { checked } = this.state;

        this.setState({
            checked: !checked,
        });

        selectItemHandler(
            item.id,
            ...selectItemHandlerArgs,
        );
    };

    label = () => {
        const { item, itemLabelProperty, isServicePoint } = this.props;

        if (isServicePoint) {
            return (
                <span>
                    {`${item[itemLabelProperty]}`}
                    {" "}
                    <br />
                    <em>{`(${item.settlementName})`}</em>
                </span>
            );
        }

        return item[itemLabelProperty];
    };

    render() {
        const { checked } = this.state;

        return (
            <FormGroup check>
                <Label check className="map-controls-checkbox">
                    <Input type="checkbox" checked={checked} onChange={this.onChange} />
                    {this.label()}
                </Label>
            </FormGroup>
        );
    }
}

GlobalFilterCheckbox.propTypes = {
    selectItemHandler: PropTypes.func.isRequired,
    item: PropTypes.objectOf(PropTypes.any).isRequired,
    selectItemHandlerArgs: PropTypes.arrayOf(PropTypes.string).isRequired,
    selectedItems: PropTypes.arrayOf(PropTypes.any).isRequired,
    itemLabelProperty: PropTypes.string.isRequired,
};
