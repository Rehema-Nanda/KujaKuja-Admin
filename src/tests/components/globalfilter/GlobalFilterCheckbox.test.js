import React from "react";
import { shallow } from "enzyme";
import GlobalFilterCheckbox from "../../../components/globalfilter/GlobalFilterCheckbox";

describe("<GlobalFilterCheckbox />", () => {
    const testProps = {
        selectedItems: ["1"],
        item: { id: "1" },
        selectItemHandler: jest.fn(),
        selectItemHandlerArgs: "Locations",
    };
    const wrapper = shallow(<GlobalFilterCheckbox {...testProps} />);

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("changes checked state when the input for checkbox changes", () => {
        const input = wrapper.find("Input");
        const state = wrapper.state().checked;
        input.simulate("change");
        expect(wrapper.state().checked).toEqual(!state);
    });
});
