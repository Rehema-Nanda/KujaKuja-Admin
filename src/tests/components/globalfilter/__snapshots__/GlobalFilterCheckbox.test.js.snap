// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalFilterCheckbox /> renders correctly 1`] = `
"<FormGroup check={true} tag=\\"div\\">
  <Label check={true} className=\\"map-controls-checkbox\\" tag=\\"label\\" widths={{...}}>
    <Input type=\\"checkbox\\" checked={true} onChange={[Function: bound ]} />
  </Label>
</FormGroup>"
`;
